File: health_care_reminder/lib/domain/entity/patient_entity.dart
// Author: Pravasta Rama - 2026

import 'package:health_care_reminder/core/enum/infusion_status_enum.dart';

import '../../core/enum/gender_enum.dart';

/// [PatientEntity] A domain entity class for patients.
/// This class represents the core attributes of a patient
/// used in the business logic layer.
class PatientEntity {
  final int? id;
  final String patientCode;
  final String name;
  final Gender gender;
  final InfusionStatusEnum? status;
  final int? tpm;
  final DateTime? startTime;
  final DateTime? endTime;

  PatientEntity({
    this.id,
    required this.patientCode,
    required this.name,
    required this.gender,
    this.status,
    this.tpm,
    this.startTime,
    this.endTime,
  });
}

File: health-care-reminder-backend/internal/core/domain/entity/patient_entity.go
package entity

import (
	"health-care-reminder/utils/enum"
	"time"
)

type PatientEntity struct {
	ID        int64
	Name      string
	Code      string
	Gender    enum.Gender
	Status    *enum.Status
	Tpm       *int64
	StartTime *time.Time
	EndTime   *time.Time
}
